frontend/src/services/api/rom.ts

```diff
@@ -123,17 +121,22 @@ async function downloadRom({
 
   const a = document.createElement("a");
   a.href = `/api/roms/${rom.id}/content/${rom.file_name}?${files_params}`;
-  a.click();
-
-  // Only connect socket if multi-file download
-  if (rom.multi && files.length > 1) {
-    if (!socket.connected) socket.connect();
-    storeDownload().add(rom.id);
-
-    // Clear download state after 60 seconds in case error/timeout
-    setTimeout(() => {
-      clearRomFromDownloads(rom);
-    }, 60 * 1000);
+  if (shareLink) {
+    navigator.clipboard.writeText(encodeURI(a.href));
+    return encodeURI(a.href);
```

Comment:
thb i really don't think this should live here, move the login to a util file for generating a download link, and use it here if you want